settings.gradle.kts

```diff
@@ -56,12 +56,16 @@ pluginManagement {
   }
 }
 
-@Suppress("UnstableApiUsage") dependencyResolutionManagement { repositories { mavenCentral() } }
+plugins {
+  id("org.gradle.toolchains.foojay-resolver-convention") version("0.8.0")
+}
```

Comment:
this will auto-download the right toolchain when one is not available; the entirety of this functionality exists within these three lines, and is optional alongside regular JVM toolchains support.

if auto-provisioning is not enabled via foojay, and no local toolchain matches the requirements needed for a build or execution, an error is thrown that halts the build.